contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2177,7 +2210,15 @@ static List
 		else
 		{
 			appendStringInfo(&query, "REVOKE dummy ON ALL TABLES IN SCHEMA dummy FROM dummy; ");
-			appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES IN SCHEMA dummy REVOKE dummy ON TABLES FROM dummy; ");
+			if (owner_other_than_dbo)
+			{
+				/* Grant ALTER DEFAULT PRIVILEGES on schema owner and dbo user. */
+				appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy IN SCHEMA dummy REVOKE dummy ON TABLES FROM dummy; ");
+				appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy IN SCHEMA dummy REVOKE dummy ON TABLES FROM dummy; ");
+				expected_stmts = 3;
+			}
+			else
+				appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES IN SCHEMA dummy REVOKE dummy ON TABLES FROM dummy; ");
```

Comment:
This is equivalent to `ALTER DEFAULT PRIVILEGES FOR ROLE *dbo* IN SCHEMA dummy REVOKE dummy ON TABLES FROM dummy;`, right ? Can we refactor/simplify the logic.

Reply:
https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2974#discussion_r1777158869